there's this test warning though 
```
390⋮    │/nix/store/2b99rpx8zwdjjqkrvk7kqgn9mxhiidjy-binutils-2.38/bin/ld: cannot find -lc: No such file or directory
 391⋮    │collect2: error: ld returned 1 exit status
 392⋮    │No support for static linking found (lacks libc.a?), skip cgo static linking test.
```

--- a/pkgs/development/compilers/go/1.17.nix
+++ b/pkgs/development/compilers/go/1.17.nix
@@ -67,9 +67,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-KZ5VrzDxVpGwFdjc+OyuckEkElaeWy7OIDYXU6RW8vk=";
   };
 
+  strictDeps = true;
   # perl is used for testing go vet
   nativeBuildInputs = [ perl which pkg-config procps ];
-  buildInputs = [ cacert pcre ];
+  buildInputs = [ cacert pcre ]
+    ++ lib.optionals stdenv.isLinux [ stdenv.cc.libc.out ]
+    ++ lib.optionals (stdenv.hostPlatform.libc == "glibc") [ stdenv.cc.libc.static ];
 
   propagatedBuildInputs = lib.optionals stdenv.isDarwin [ xcbuild ];
 

--- a/pkgs/development/compilers/go/1.18.nix
+++ b/pkgs/development/compilers/go/1.18.nix
@@ -67,9 +67,12 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-LETQPqLDQJITerkZumAvLCYaA40I60aFKKPzoo5WZ+I=";
   };
 
+  strictDeps = true;
   # perl is used for testing go vet
   nativeBuildInputs = [ perl which pkg-config procps ];
-  buildInputs = [ cacert pcre ];
+  buildInputs = [ cacert pcre ]
+    ++ lib.optionals stdenv.isLinux [ stdenv.cc.libc.out ]
+    ++ lib.optionals (stdenv.hostPlatform.libc == "glibc") [ stdenv.cc.libc.static ];
 
   propagatedBuildInputs = lib.optionals stdenv.isDarwin [ xcbuild ];
 

